src/screens/CostsEntryScreen/CostsEntryScreen.tsx

```diff
@@ -1,15 +1,162 @@
 
 type Props = NavigationData<'CostsEntry'>;
 
+export interface CostFormValues {
+  amount: string;
+  amountKind: AmountKind;
+  currency: Currency | '';
+  vatRate: VatRate | '';
+  purchaseDate: Date;
+}
+
 const CostsEntryScreen: React.FC<Props> = () => {
+  const { t } = useAppTranslation();
+
+  const CostEntrySchema = Yup.object<Partial<CostFormValues>>({
+    amount: Yup.string().required(t('Required')),
+    amountKind: Yup.string<AmountKind>().required(t('Required')),
+    currency: Yup.string<Currency>().required(t('Required')),
+    vatRate: Yup.number<VatRate>().required(t('Required')),
+    purchaseDate: Yup.date().required(t('Required')),
+  });
+  const initialValues: CostFormValues = {
+    amount: '',
+    amountKind: AmountKind.NET,
+    // TODO: fill currency with base account currency in app?
+    currency: '',
+    vatRate: '',
+    purchaseDate: new Date('2021-01-01'),
+  };
+
   return (
-    <View>
-      <Text>Costs Entry</Text>
-    </View>
+    <ScrollView
+      keyboardShouldPersistTaps="handled"
+      style={{ backgroundColor: '#ffffff' }}
+    >
+      <ScreenMainStyledView>
+        <Formik
+          initialValues={initialValues}
+          validationSchema={CostEntrySchema}
+          onSubmit={({
+            amount,
+            amountKind,
+            currency,
+            vatRate,
+            purchaseDate,
+          }: CostFormValues) => {
+            if (!currency || !vatRate) return;
+
+            const item = new CostElement(
+              +amount,
+              amountKind,
+              currency,
+              vatRate,
+              purchaseDate
+            );
+
+            // eslint-disable-next-line no-console
+            console.log(item);
+            // TODO: send cost item to firebase/SQLite store
+          }}
+        >
+          {({
+            handleChange,
+            handleBlur,
+            handleSubmit,
+            setFieldTouched,
+            setFieldValue,
+            values,
+            errors,
+            touched,
+            isValid,
+          }) => (
+            <StyledWideContainer>
+              <TextField
+                value={values.amount}
+                placeholder={t('Amount')}
+                onChangeText={handleChange('amount')}
+                onBlur={handleBlur('amount')}
+                error={errors.amount}
+                touched={touched.amount}
+                keyboardType="decimal-pad"
+              />
+              <SelectField
+                value={values.currency}
+                placeholder={t('Currency')}
+                error={errors.currency}
+                touched={touched.currency}
+                onValueChange={handleChange('currency')}
+                onClose={() => setFieldTouched('currency')}
+                selectorPlaceholder={t('Select a currency')}
+                items={Object.values(Currency).map(item => ({
+                  label: item,
+                  value: item,
+                }))}
```

Comment:
Maybe it's better to call this prop `options`? Much more common I think.

Reply:
There's some idea how to change a default name of `items` prop to `options`:
```
export interface SelectFieldProps extends Omit<PickerSelectProps, 'items'> {
  options: PickerSelectProps['items'];
}
```

Now I'm not sure about the change, but maybe it will make it consistent with other components (for example for `SegmentRadioField` we've got `radioOptions` prop).

Reply:
I knew something like this exist but I didn't find it :smile: thanks! I will use it